sql-plugin/src/main/scala/com/nvidia/spark/rapids/literals.scala

```diff
@@ -233,3 +234,32 @@ case class GpuLiteral (value: Any, dataType: DataType) extends GpuLeafExpression
 
   override def columnarEval(batch: ColumnarBatch): Any = value
 }
+
+class LiteralExprMeta(
+    lit: Literal,
+    conf: RapidsConf,
+    p: Option[RapidsMeta[_, _, _]],
+    r: DataFromReplacementRule) extends ExprMeta[Literal](lit, conf, p, r) {
+
+  override def convertToGpu(): GpuExpression = {
+    lit.dataType match {
+      // NOTICE: There is a temporary transformation from Literal(ArrayType(BaseType)) into
+      // CreateArray(Literal(BaseType):_*). The transformation is a walkaround support for Literal
+      // of ArrayData under GPU runtime, because cuDF scalar doesn't support nested types.
+      case ArrayType(baseType, _) =>
+        val litArray = lit.value.asInstanceOf[ArrayData]
+          .array.map(GpuLiteral(_, baseType))
+        GpuCreateArray(litArray, useStringTypeWhenEmpty = false)
```

Comment:
I am a bit nervous about this change.  Because cudf does not support literal arrays I would want to see a lot more testing of literal arrays in other places in queries, not just explode.

Reply:
I've added some tests for Literal(ArrayData) in `ProjectExprSuite`.

Reply:
OK Can we have a follow on issue to remove this once cudf supports scalar arrays? Which should be fairly soon.

Reply:
I filed an issue for this (#1902).